inception-ui-kb/src/main/java/de/tudarmstadt/ukp/inception/ui/kb/value/editor/IRIValuePresenter.java

```diff
@@ -17,44 +17,95 @@
+import de.tudarmstadt.ukp.inception.kb.model.KnowledgeBase;
 import de.tudarmstadt.ukp.inception.ui.kb.event.AjaxConceptSelectionEvent;
+import de.tudarmstadt.ukp.inception.ui.kb.event.AjaxInstanceSelectionEvent;
+import de.tudarmstadt.ukp.inception.ui.kb.event.AjaxPropertySelectionEvent;
+
 
 public class IRIValuePresenter
     extends ValuePresenter
 {
     private static final long serialVersionUID = -2127902473859929221L;
 
-    public IRIValuePresenter(String id, IModel<KBStatement> aModel)
+    private static final Logger LOG = LoggerFactory.getLogger(IRIValuePresenter.class);
+
+    @SpringBean KnowledgeBaseService kbService;
+
+    private IModel<KnowledgeBase> kbModel;
+
+    public IRIValuePresenter(String id, IModel<KBStatement> aModel, IModel<KnowledgeBase> aKBModel)
     {
         super(id, aModel);
+        kbModel = aKBModel;
 
-        LambdaAjaxLink link = new LambdaAjaxLink("link", this::actionIRILinkClicked);
-        link.add(new Label("label", LambdaModel.of(this::getLabel)));
+        LambdaAjaxLink link = new LambdaAjaxLink("link",
+            t -> actionIRILinkClicked(t, getKBObject()));
+        link.add(new Label("label", Model.of(getLabel(getKBObject()))));
         add(link);
     }
 
-    private String getLabel()
+    Optional<KBObject> getKBObject() {
+        Object stmtValue = getModelObject().getValue();
+        Optional<KBObject> kbObject = kbService
+            .readKBIdentifier(kbModel.getObject(), stmtValue.toString());
+        if (kbObject.isPresent()) {
+            return kbObject;
+        }
+        return Optional.empty();
+    }
+
+    private String getLabel(Optional<KBObject> aKbObject)
     {
-        KBStatement stmt = getModelObject();
-        return ((IRI) stmt.getValue()).getLocalName();
+        if (aKbObject.isPresent()) {
+            return aKbObject.get().getUiLabel();
+        }
+        return ((IRI) getModelObject().getValue()).getLocalName();
     }
 
-    private void actionIRILinkClicked(AjaxRequestTarget target)
+    private void actionIRILinkClicked(AjaxRequestTarget aTarget, Optional<KBObject> aKbObject)
     {
-        // TODO need to know what the IRI refers to - concept, property, both???
-        KBStatement stmt = getModelObject();
-        KBHandle selectedConcept = new KBHandle(((IRI) stmt.getValue()).toString());
-        send(getPage(), Broadcast.BREADTH,
-                new AjaxConceptSelectionEvent(target, selectedConcept, true));
+        if (aKbObject.isPresent()) {
+            KBObject kbObject = aKbObject.get();
+            if (kbObject instanceof KBConcept) {
+                send(getPage(), Broadcast.BREADTH,
+                    new AjaxConceptSelectionEvent(aTarget, KBHandle.of(kbObject), true));
+            }
+            else if (kbObject instanceof KBInstance) {
+                send(getPage(), Broadcast.BREADTH,
+                    new AjaxInstanceSelectionEvent(aTarget, KBHandle.of(kbObject)));
+            }
+            else {
```

Comment:
Could we do a `instanceof KBProperty` here and throw an exception if the KBObject is an unknown type of instance?